testcontainers/src/core/containers/sync_container.rs

```diff
@@ -257,17 +255,12 @@ where
 
 impl<I: Image> Drop for Container<I> {
     fn drop(&mut self) {
-        if let Some(active) = self.inner.take() {
-            active.runtime.block_on(async {
-                match active.async_impl.docker_client().config.command() {
-                    env::Command::Remove => {
-                        if let Err(e) = active.async_impl.rm().await {
```

Comment:
This is why containers are always removed as reported in https://github.com/testcontainers/testcontainers-rs/issues/742#issuecomment-3451607141

I'm using sync Containers and its Drop impl isn't properly forwarded to the async impl.